### PR TITLE
Implement changes to Blazor Admin Support page for AB#13978

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#13978](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13978)

## Description

- Implement changes as per [AB#13931](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/13931) for Blazor Admin Support
- Users will be calling  support users api to retrieve users
- Within each user there will be another call to messaging verifications to return associated verifications
- Implemented validation for SMS and Email
- Fixed admin server model code to remove unused using statement
- Added verification data to seed data script

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [x] Not Required

## UI Changes

**Exception Message during Messaging Verification call:**

Screen Shot 2022-10-12 at 2 19 11 PM

**Email Required:**

Screen Shot 2022-10-12 at 3 06 00 PM

**Email must be minimum 5 characters:**

Screen Shot 2022-10-12 at 2 00 13 PM

**Email Success:**

Screen Shot 2022-10-12 at 2 00 41 PM

**SMS required:**

Screen Shot 2022-10-12 at 3 06 12 PM

**SMS must be minimum 5 characters:**

Screen Shot 2022-10-12 at 2 01 27 PM

**SMS success:**

Screen Shot 2022-10-12 at 2 01 42 PM

**HDID required:**

Screen Shot 2022-10-12 at 2 01 57 PM

**HDID success:**

Screen Shot 2022-10-12 at 2 02 56 PM

**HDID client registry error:**

Screen Shot 2022-10-12 at 2 03 14 PM

**HDID client registry warning:**

Screen Shot 2022-10-12 at 12 12 29 PM

**PHN invalid:**

Screen Shot 2022-10-12 at 2 03 48 PM

**PHN success:**

Screen Shot 2022-10-12 at 2 04 00 PM

**Multiple users and multiple verifications:**

Screen Shot 2022-10-12 at 1 31 23 PM

**No Verification:**

Screen Shot 2022-10-12 at 1 38 09 PM

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by mk08sh — COMMENTED** [review]
👍

**Review by rayking29 — APPROVED** [review]

File: Apps/Admin/Client/Store/SupportUser/SupportUserReducers.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        public static SupportUserState ReduceResetStateAction(SupportUserState state)
+        {
+            return state with
+            {
+                IsLoading = false,
+                Result = null,
+                Error = null,
+                Data = null,
+            };
+        }
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Store/SupportUser/SupportUserReducers.cs:92]: The WarningMessage property should also be reset.

File: Apps/Admin/Client/Pages/SupportPage.razor [hunk 2]
```diff
@@ -38,56 +43,62 @@
     </MudGrid>
 </MudForm>
 
-<HgBannerFeedback Severity="@Severity.Info" IsEnabled="HasWarning" TResetAction="MessageVerificationActions.ResetStateAction" DataTestId="banner-feedback-warning-message">
-    @MessageVerificationState.Value.WarningMessage
+<HgBannerFeedback Severity="@Severity.Info" IsEnabled="HasHasSupportUsersWarning" TResetAction="SupportUserActions.ResetStateAction" DataTestId="banner-feedback-warning-message">
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/SupportPage.razor:46]: Typo in variable name `HasHasSupportUsersWarning` (matched in code-behind).

**Review by MikeLyttle — COMMENTED** [review]
Some minor issues noted but looks good otherwise.

File: Apps/Admin/Client/Pages/SupportPage.razor (hunk 2 above)
Comment (BrianMaki) [reply Apps/Admin/Client/Pages/SupportPage.razor:46]: Fixed.

File: Apps/Admin/Client/Store/SupportUser/SupportUserReducers.cs (hunk 1 above)
Comment (BrianMaki) [reply Apps/Admin/Client/Store/SupportUser/SupportUserReducers.cs:92]: Fixed.

**Pull request merged**